chore(surveys): add branching logic docs (#8886)
PostHog/posthog.com: contents/docs/surveys/branching-logic.mdx

---

> title: Branching logic

Reviewer: I think this is a better title. "Branching" adds new jargon
```suggestion
title: Conditional questions
```

Author: agree!

sidebar: Docs
showTitle: true
availability:
  free: none
  selfServe: full
  enterprise: full
---

> It is useful to be able to display survey questions conditionally. For example, you may want to ask certain questions based on the responses received. Branching logic allows you to do this.

Reviewer: ```suggestion
It is useful to be able to display survey questions conditionally. For example, you may want to ask certain questions based on the responses received. PostHog offers several ways to do this:
```

> ## Simple branching

Reviewer: Since `After this question, go to:` is also in the form creation, can you also add a heading for it in /docs/surveys/creating-surveys? You can link to this doc from there

For every question type, you can now define the next step to navigate to after a question is completed. Click on __After this question, go to:__, then choose the next step you want to transition to. The options are:

- Next question
- Any specific question
- Confirmation message (if present) or End

> After the given question is completed, the chosen next step will be followed.

Reviewer: I dont understand how this works. Why would you not just proceed to the next logical step e.g. next question / end of survey? What does this enable you to do?

Author: Let's say you have: q1, q2, q3, confirmation message.
You are at q1. Depending on how q1 is answered, you will navigate either to q2 or q3.

q2 and q3 should both continue to the confirmation message. For q2 you will need to say this explicitly, otherwise it would continue to q3 by default. Simple branching allows you to specify this ("from q2 always go to the confirmation message")

> ## Response-based branching

Reviewer: ```suggestion
## Response-based conditions
```

> For the __Single choice__ and __Rating__ question types, you can define the next step based on the response received. Click on __After this question, go to:__, and choose the option __Specific question based on answer__. Depending on your question type, this will present the following options.

Reviewer: ```suggestion
For the **single choice** and **rating** question types, you can define the next step based on the response received. 

Click on **After this question, go to:**, and then choose **Specific question based on answer**. Depending on your question type, this will present the following options:
```

> For the __Single choice__ question type, you can specify the next step based on the selected answer.

Reviewer: ```suggestion
1. For the __Single choice__ question type, you can specify the next step based on the selected answer.
```

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/surveys_branching_singlechoice_8053dd1700.png" 
    imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/surveys_branching_singlechoice_dark_0ec63f974a.png"
    alt="Survey single choice branching" 
    classes="rounded"
/>

> For the __Rating__ question type, you can specify the next step based on the response bucket received. For example, for a rating question with a scale from 1 to 5, the responses will be bucketed into categories: _negative_, _neutral_, and _positive_. The branching will be executed based on which bucket the individual response falls into.

Reviewer: ```suggestion
2. For the __Rating__ question type, you can specify the next step based on the response bucket received. For example, for a rating question with a scale from 1 to 5, the responses will be bucketed into categories: _negative_, _neutral_, and _positive_. The conditions are executed based on which bucket the individual response falls into.
```

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_09_29_210ff7ced4.png" 
    imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_10_37_7804a733b4.png"
    alt="Survey rating branching" 
    classes="rounded"
/>

> __Note__: If you intend for a question to be the last one, but it is not the final question in the list, you need to explicitly set the branching to the __Confirmation message__ (if present) or __End__. By default, a question will branch to the next question unless specified otherwise.

Reviewer 2: ```suggestion
> __Note__: If you intend for a question to be the last one, but it is not the final question in the list, you need to explicitly set the branching to the __Confirmation message__ (if present) or __End__. By default, a question will branch to the next question unless specified otherwise.
```

> __Note__: If you intend for a question to be the last one, but it is not the final question in the list, you need to explicitly set the branching to the __Confirmation message__ (if present) or __End__. By default, a question will branch to the next question unless specified otherwise.

Reviewer: ```suggestion
By default, a question proceed to the next question in the survey unless specified otherwise. If you want a question to be the last one, but it is not the final question in the list, you need to explicitly set the value for **After this question go to ** to __Confirmation message__ (if present) or __End__..
```
